### PR TITLE
Merge MaterialX:main 1.38.5 to adsk_contrib 

### PR DESCRIPTION
**Pull request by ashwinbhat**
Please see main changelog for details. One of the main changes is how libraries are handled that impacts MaterialXRuntime, OGSFX, OGSXML and Arnold Shader Generators.
Key changes to Autodesk fork:
- OGSFX/OGSXML tests have been updated
- **disabled** MaterialXRuntime tests.
- **disabled** adsk_shader tests since they use textures from `libraries` which is not recommended. Needs further discussion. 
-  The Arnold tests pass but they emit the following warnings. Needs further investigation.
```
77: --- Wrote temp arnold mtlx file to: F:\source\MaterialX-adsk-fork-dev\_build_test\bin\resources\Materials\TestSuite\stdlib\version\multiple_version_test.v2_implicit_out_arnold.xml
77: Render: Arnold TestSuite completed in 133.292s
77: ===============================================================================
77: All tests passed (1697 assertions in 1 test case)
77:
77/77 Test #77: MaterialXTest_Render_Arnold_TestSuite ........................   Passed  133.43 sec



```

### CODE REVIEW

**Review by francis-wangfr — COMMENTED** [review]
Hi Ashwin, it looks the merge is huge, in that case, it seems test cases are essential for us to know how correct the merging. I don't have chance to sync and test. How about test cases failure you mentioned about Arnold? If they could be fixed or related reviewer think they are not critical that's great to go I think.

**Review by francis-wangfr — APPROVED** [review]
LGTM as Orn mentioned it is OK to disable the Arnold runtime test cases.

**Review by JGamache-autodesk — APPROVED** [review]

**Review by nicolassavva-autodesk — APPROVED** [review]
Some of the documentation updates have tripped Codacy.
I believe we can just mark it as resolved.
From what I gather from the history we also pulled in 1.38.6 commits
so the PR description might not be entirely accurate here

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # MaterialX Version
 set(MATERIALX_MAJOR_VERSION 1)
 set(MATERIALX_MINOR_VERSION 38)
-set(MATERIALX_BUILD_VERSION 5)
+set(MATERIALX_BUILD_VERSION 6)
```
Comment (nicolassavva-autodesk) [inline CMakeLists.txt:4]: The 1.38.6 version bump already happened

**Pull request merged**